### PR TITLE
fix: get input data based on address type

### PR DESCRIPTION
**Pull request by bucko13**
Some eslint fixes got caught in this but basically the input data for adding to a psbt needs to be a bit smarter about how it gets the input data for ledger v2 which is pickier about when it requires nonWitnessUtxo vs witnessUtxo data. (See [here](https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki#simple-signer-algorithm) from the bip174 docs).

Unfortunately while this creates a valid PSBT that ledger will sign, the signatures created after getting converted to psbtv2 and getting passed to the ledger are still wrong so there's still some tweaking left to do. 

Can test with: https://github.com/unchained-capital/caravan/pull/294

### CODE REVIEW

File: src/psbt.js [hunk 1]
```diff
@@ -200,13 +202,20 @@ export function psbtInputFormatter(input) {
...
-  const utxoToVerify = isSegWit ? {witnessUtxo} : {nonWitnessUtxo};
+  const addressType = multisigAddressType(input.multisig);
+
+  // The PSBT class in bitcoinjs-lib won't accept both nonWitnessUxo and witnessUtxo
+  // but ledger v2 needs nonWitnessUtxo data in the PSBT for any transactions that aren't
+  // native segwit. So we need to pick which one to give based on the address type
+  // https://github.com/bitcoinjs/bitcoinjs-lib/issues/1595
+  const utxoToVerify =
+    addressType === P2WSH ? { witnessUtxo } : { nonWitnessUtxo };
+
```
Comment (bucko13) [inline src/psbt.js:218]: this is basically the only relevant change

**Review by Shadouts — APPROVED** [review]

**Review by Shadouts — CHANGES_REQUESTED** [review]
I know we talked about this out-of-band, but there are still some signature test failures on this which is to be expected since the psbt contents are changing.

**Comment by Shadouts** [comment]
I went to go see about the sigs myself, but I'm still getting invalid data error `0x6a80`

**Comment by bucko13** [comment]
@Shadouts thats weird…

**Comment by bucko13** [comment]
So, the correct behavior for this is that if it's `P2SH-P2WSH` then BOTH nonWitnessUtxo and witnessUtxo should be included in the PSBT. Unfortunately bitcoinjs-lib doesn't allow you to do this though. I've filed a [bug](https://github.com/bitcoinjs/bitcoinjs-lib/issues/1894). Hopefully it'll be addressed and we can fix the input serialization in the correct way such that signatures are created correctly. Without this change, ledger will fail b/c it has incorrect data. With this change, ledger will succeed but will produce incorrect signatures so it's not quite ready for primetime.

**Comment by bucko13** [comment]
Not required anymore for now. Closing.

**Pull request closed without merging**